src/main/java/org/ibp/api/java/impl/middleware/germplasm/GermplasmListServiceImpl.java

```diff
@@ -467,6 +487,61 @@ private void processEntries(final GermplasmListGeneratorDTO request, final Strin
 
 	}
 
+	private void processEntriesForUpdate(final GermplasmListGeneratorDTO request) {
+
+		BaseValidator.checkNotNull(request.getId(), "error.germplasmlist.importupdates.listid.mandatory");
+
+		final Map<Integer, Variable> entryDetailVariablesById = this.extractVariableIds(request);
+
+		for (final GermplasmListGeneratorDTO.GermplasmEntryDTO entry : request.getEntries()) {
+			if (entry.getEntryNo() == null) {
+				throw new ApiValidationException("", "error.germplasmlist.importupdates.entryno.mandatory");
+			}
+
+			this.processEntryDetails(entry.getData(), entryDetailVariablesById);
+		}
+	}
+
+	private Map<Integer, Variable> extractVariableIds(final GermplasmListGeneratorDTO request) {
+		final List<Integer> entryDetailVariableIds = request.getEntries().stream()
+			.flatMap(germplasmEntryDTO -> germplasmEntryDTO.getData().keySet().stream())
+			.collect(Collectors.toList());
+
+		final VariableFilter filter = new VariableFilter();
+		entryDetailVariableIds.forEach(filter::addVariableId);
+		return !isEmpty(entryDetailVariableIds)
+			? this.ontologyVariableService.getVariablesWithFilterById(filter)
+			.entrySet().stream().collect(toMap(entry -> entry.getValue().getId(), Map.Entry::getValue))
+			: Collections.emptyMap();
+	}
+
+	private void processEntryDetails(
+		final Map<Integer, GermplasmListObservationDto> data,
+		final Map<Integer, Variable> entryDetailVariablesById
+	) {
+		// complete variable info based on request
+		for (final Map.Entry<Integer, GermplasmListObservationDto> entryDetailSet : data.entrySet()) {
+
+			final GermplasmListObservationDto entryDetail = entryDetailSet.getValue();
+			final Variable variable = entryDetailVariablesById.get(entryDetailSet.getKey());
+			BaseValidator.checkNotNull(variable, "germplasm.list.variable.does.not.exist");
+			final String value = entryDetail.getValue();
+			this.validateVariableDataTypeValue(variable, value);
+
+			entryDetail.setVariableId(variable.getId());
+			entryDetail.setcValueId(VariableValueUtil.resolveCategoricalValueId(variable, value));
+		}
+	}
+
+	private void validateVariableDataTypeValue(final Variable variable, final String value) {
+		if (!VariableValueUtil.isValidAttributeValue(variable, value)) {
+			throw new ApiRequestValidationException("invalid.variable.value.with.param", new String[] {
+				"variable: " + variable.getName() + ", value: " + value
+			});
+		}
+		checkArgument(value.length() <= 255, "text.field.max.length", new String[] {"entry detail value", "255"});
```

Comment:
same here, in case we want to reuse the property I think we can use the attribute name instead of a custom text so at least it is associated the the body we are expecting